### PR TITLE
[macOS] Flavors project throws `no flavor specified` for creating a project.

### LINKED ISSUE #113978: [flutter_tools] `flutter create --platforms=<>` on a project with flavors will crash
For example:

```
[   +1 ms] The Xcode project defines schemes: free, paid
[   +5 ms] "flutter create" took 2,080ms.
[   +2 ms] You must specify a --flavor option to select one of the available schemes.
[        ] 
           #0      throwToolExit (package:flutter_tools/src/base/common.dart:10:3)
           #1      XcodeProjectInfo.reportFlavorNotFoundAndExit
           (package:flutter_tools/src/ios/xcodeproj.dart:489:7)
           #2      IosProject.buildSettingsForBuildInfo
(package:flutter_tools/src/xcode_project.dart:270:12)
           <asynchronous suspension>
           #3      IosProject._parseProductBundleIdentifier
           (package:flutter_tools/src/xcode_project.dart:202:51)
           <asynchronous suspension>
           #4      IosProject.productBundleIdentifier
(package:flutter_tools/src/xcode_project.dart:179:41)
           <asynchronous suspension>
           #5      FlutterProject.organizationNames (package:flutter_tools/src/project.dart:141:40)
           <asynchronous suspension>
           #6      CreateBase.getOrganization
(package:flutter_tools/src/commands/create_base.dart:259:49)
           <asynchronous suspension>
           #7      CreateCommand.runCommand (package:flutter_tools/src/commands/create.dart:243:33)
           <asynchronous suspension>
           #8      FlutterCommand.run.<anonymous closure>
           (package:flutter_tools/src/runner/flutter_command.dart:1244:27)
           <asynchronous suspension>
           #9      AppContext.run.<anonymous closure>
(package:flutter_tools/src/base/context.dart:150:19)
           <asynchronous suspension>
           #10     CommandRunner.runCommand (package:args/command_runner.dart:209:13)
           <asynchronous suspension>
           #11     FlutterCommandRunner.runCommand.<anonymous closure>
           (package:flutter_tools/src/runner/flutter_command_runner.dart:283:9)
           <asynchronous suspension>
           #12     AppContext.run.<anonymous closure>
(package:flutter_tools/src/base/context.dart:150:19)
           <asynchronous suspension>
           #13     FlutterCommandRunner.runCommand
           (package:flutter_tools/src/runner/flutter_command_runner.dart:229:5)
           <asynchronous suspension>
           #14     run.<anonymous closure>.<anonymous closure>
(package:flutter_tools/runner.dart:64:9)
           <asynchronous suspension>
           #15     AppContext.run.<anonymous closure>
(package:flutter_tools/src/base/context.dart:150:19)
           <asynchronous suspension>
           #16     main (package:flutter_tools/executable.dart:91:3)
           <asynchronous suspension>
```

Note that `flutter create` does not implement the `--flavors` flag, so this comment is misleading.

### PR DESCRIPTION
**Pull request by a-wallen**
`--flavors` is not exposed in Flutter create, but in the process of parsing `buildInfo` when creating a new Flutter project, a tool exit is thrown if flavors are found, but no flavors are specified at the command line.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter_tools/test/general.shard/project_test.dart [hunk 1]
```diff
@@ -509,6 +509,17 @@ apply plugin: 'kotlin-android'
         expect(await project.ios.productBundleIdentifier(null), 'io.flutter.someProject.suffix');
       });
 
+      testWithMocks('foo bar', () async {
```
Comment (christopherfujino) [inline packages/flutter_tools/test/general.shard/project_test.dart:512]: please update test name

**Review by christopherfujino — APPROVED** [review]
LGTM

**Review by jmagman — COMMENTED** [review]
Catching `ToolExit`s seems like kind of a hack.  Could you instead make flavors work in this case and support the `--flavor` flag? https://github.com/flutter/flutter/issues/99607#issuecomment-1085164018
Or throw something else?

**Comment by a-wallen** [comment]
@jmagman is it possible for `flutter create` to support a `--flavor` option? I was under the impression that flavors needed to be configured with platform tooling.

**Comment by christopherfujino** [comment]
> @jmagman is it possible for `flutter create` to support a `--flavor` option? I was under the impression that flavors needed to be configured with platform tooling.

I mean, we're autogenerating the default platform config, there's no reason we can't generate it with the flavors configured. That would be more work though.

As for throwing something else, I considered that, although we'd likely want to catch that other exception from all the OTHER callsites and wrap it in a tool exit again.

**Comment by a-wallen** [comment]
> I mean, we're autogenerating the default platform config, there's no reason we can't generate it with the flavors configured. That would be more work though.

I think that would be useful, but that's a separate concern. The `--flavor` should retain its semantics across flutter commands. I don't think those semantics are applicable to `flutter create`. 

I'd be happy to try and add something where flavors from `iOS` are copied to `macOS` and visa versa. Do you think it's possible @jmagman?

**Review by jmagman — APPROVED** [review]
Hm, I guess with `create` adding a `--flavors` flag does imply those flavors will be created for you.  I was thinking about the `install` case where it makes more sense and this same bundle ID code needs to be flavor-aware.

This is fine.

**Comment by a-wallen** [comment]
@jmagman I'm working on the [install case](https://github.com/flutter/flutter/issues/99607). Thanks for your help!

**Comment by quetool** [comment]
Hello! I suppose this is still not on stable channel, right?

**Pull request merged**